IPNS record migration cron job can be halted by an intermittent connection error

From the logs this morning:

```
file:///home/runner/work/web3.storage/web3.storage/node_modules/@web-std/fetch/src/index.js:104
			reject(new FetchError(`request to ${request.url} failed, reason: ${err.message}`, 'system', err));
			       ^

FetchError: request to https://name.web3.storage/name/k51qzi5uqu5dlfklycdliyp1jxu6ke895vh5tfwk2uunlsarxocpf6bam79055 failed, reason: read ECONNRESET
    at ClientRequest.<anonymous> (file:///home/runner/work/web3.storage/web3.storage/node_modules/@web-std/fetch/src/index.js:104:11)
    at ClientRequest.emit (node:events:539:35)
    at TLSSocket.socketErrorListener (node:_http_client:454:9)
    at TLSSocket.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  type: 'system',
  errno: 'ECONNRESET',
  code: 'ECONNRESET',
  erroredSysCall: 'read'
}
npm ERR! Lifecycle script `start:names` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: @web3-storage/cron@1.5.7 
npm ERR!   at location: /home/runner/work/web3.storage/web3.storage/packages/cron 
Error: Process completed with exit code 1.
```

If it's straightforward, it would be nice to wrap this in a quick retry.

See: https://github.com/web3-storage/web3.storage/runs/7870949708?check_suite_focus=true

--- a/packages/cron/src/jobs/names.js
+++ b/packages/cron/src/jobs/names.js
@@ -1,5 +1,5 @@
 import debug from 'debug'
-import retry from 'p-retry'
+import { pRetry, retry } from 'p-retry'
 import throttledQueue from 'throttled-queue'
 
 const MAX_REQUEST_PAGE = 10
@@ -64,12 +64,16 @@ async function postKeyW3Name (name, w3nameApiURL = '') {
 
   if (!w3nameApiURL) { return }
 
-  const res = await fetch(new URL(`/name/${key}`, w3nameApiURL), {
-    method: 'POST',
-    body: name.record
-  })
+  // Very occasionally this gets a FetchError due to the connection being reset, which
+  // if not caught causes the whole cron job to fail. Hence
+  pRetry(async () => {
+    const res = await fetch(new URL(`/name/${key}`, w3nameApiURL), {
+      method: 'POST',
+      body: name.record
+    })
 
-  if (log.enabled) {
-    console.log(res.status, res.statusText, await res.text())
-  }
+    if (log.enabled) {
+      console.log(res.status, res.statusText, await res.text())
+    }
+  }, { retries: 1, onFailedAttempt: log })
 }

--- a/packages/cron/src/jobs/names.js
+++ b/packages/cron/src/jobs/names.js
@@ -1,5 +1,5 @@
 import debug from 'debug'
-import { pRetry, retry } from 'p-retry'
+import { pRetry } from 'p-retry'
 import throttledQueue from 'throttled-queue'
 
 const MAX_REQUEST_PAGE = 10
@@ -24,7 +24,7 @@ export async function postNamesToW3name ({ env, db }) {
       console.log('Query page', from, to)
     }
 
-    const queryRes = await retry(() => db.listNames({ from, to }), { onFailedAttempt: log })
+    const queryRes = await pRetry(async () => await db.listNames({ from, to }), { onFailedAttempt: log })
 
     if (queryRes.length > 0) {
       await Promise.all(queryRes.map((name) => throttle(() => postKeyW3Name(name, w3nameApiURL))))

--- a/packages/cron/src/jobs/names.js
+++ b/packages/cron/src/jobs/names.js
@@ -66,7 +66,7 @@ async function postKeyW3Name (name, w3nameApiURL = '') {
 
   // Very occasionally this gets a FetchError due to the connection being reset, which
   // if not caught causes the whole cron job to fail. Hence
-  pRetry(async () => {
+  await pRetry(async () => {
     const res = await fetch(new URL(`/name/${key}`, w3nameApiURL), {
       method: 'POST',
       body: name.record

